nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/JsonTreeReader.java

```diff
@@ -59,32 +61,62 @@
         + "If the schema that is configured contains a field that is not present in the JSON, a null value will be used. If the JSON contains "
         + "a field that is not present in the schema, that field will be skipped. "
-    + "See the Usage of the Controller Service for more information and examples.")
+        + "See the Usage of the Controller Service for more information and examples.")
 @SeeAlso(JsonPathReader.class)
 public class JsonTreeReader extends SchemaRegistryService implements RecordReaderFactory {
 
     private volatile String dateFormat;
     private volatile String timeFormat;
     private volatile String timestampFormat;
+    private volatile String startingFieldName;
+    private volatile StartingFieldStrategy startingFieldStrategy;
+
+    public static final PropertyDescriptor STARTING_FIELD_STRATEGY = new PropertyDescriptor.Builder()
+            .name("starting-field-strategy")
+            .displayName("Starting Field Strategy")
+            .description("Start processing from the root node or from a specified nested node.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .defaultValue(StartingFieldStrategy.ROOT_NODE.name())
+            .allowableValues(
+                    Arrays.stream(StartingFieldStrategy.values()).map(startingStrategy ->
+                            new AllowableValue(startingStrategy.name(), startingStrategy.name(), startingStrategy.getDescription())
+                    ).toArray(AllowableValue[]::new))
+            .build();
+
+
+    public static final PropertyDescriptor STARTING_FIELD_NAME = new PropertyDescriptor.Builder()
+            .name("starting-field-name")
+            .displayName("Starting Field Name")
+            .description("Skips forward to the given nested JSON field (array or object) to begin processing.")
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .defaultValue(null)
+            .dependsOn(STARTING_FIELD_STRATEGY, StartingFieldStrategy.NESTED_NODE.name())
+            .build();
 
     @Override
     protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
         final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
         properties.add(new PropertyDescriptor.Builder()
-            .fromPropertyDescriptor(SCHEMA_CACHE)
-            .dependsOn(SCHEMA_ACCESS_STRATEGY, INFER_SCHEMA)
-            .build());
+                .fromPropertyDescriptor(SCHEMA_CACHE)
+                .dependsOn(SCHEMA_ACCESS_STRATEGY, INFER_SCHEMA)
+                .build());
         properties.add(DateTimeUtils.DATE_FORMAT);
         properties.add(DateTimeUtils.TIME_FORMAT);
         properties.add(DateTimeUtils.TIMESTAMP_FORMAT);
+        properties.add(STARTING_FIELD_STRATEGY);
+        properties.add(STARTING_FIELD_NAME);
```

Comment:
Recommend moving the required field before the optional date fields:
```suggestion
        properties.add(STARTING_FIELD_STRATEGY);
        properties.add(STARTING_FIELD_NAME);
        properties.add(DateTimeUtils.DATE_FORMAT);
        properties.add(DateTimeUtils.TIME_FORMAT);
        properties.add(DateTimeUtils.TIMESTAMP_FORMAT);
```